lib/matplotlib/cbook.py

```diff
@@ -1505,6 +1505,14 @@ def issubclass_safe(x, klass):
 
 def safe_masked_invalid(x, copy=False):
     x = np.array(x, subok=True, copy=copy)
+    if not x.dtype.isnative:
+        if copy:
+            # if a copy, swap in place
+            x.byteswap(True)
+            x = x.newbyteorder()
+        else:
+            # otherwise make a swapped copy
+            x = x.byteswap(False).newbyteorder()
```

Comment:
agree with @efiring, but with the additional note to make sure the
reasoning is well-explained. Seeing the "copy" variable as an argument to
byteswap() was very confusing to me, because I kept thinking that the
intent was to do a copy, which I knew was wrong. My brain almost
segfaulted...

On Sat, Jul 2, 2016 at 2:20 AM, Eric Firing <notifications@github.com>
wrote:

> In lib/matplotlib/cbook.py
> <https://github.com/matplotlib/matplotlib/pull/6674#discussion_r69373645>:
>
> > @@ -1505,6 +1505,14 @@ def issubclass_safe(x, klass):
> >
> >  def safe_masked_invalid(x, copy=False):
> >      x = np.array(x, subok=True, copy=copy)
> > +    if not x.dtype.isnative:
> > +        if copy:
> > +            # if a copy, swap in place
> > +            x.byteswap(True)
> > +            x = x.newbyteorder()
> > +        else:
> > +            # otherwise make a swapped copy
> > +            x = x.byteswap(False).newbyteorder()
>
> Why is this if/else needed? I think that x =
> x.byteswap(copy).newbyteorder() will suffice. The byteswap method returns
> a new array if copy is False and a view into the (modified) original if
> copy (inplace) is True.
>
> In [31]: xm = np.ma.ones((2,))
>
> In [32]: xm1 = xm.byteswap(False).newbyteorder()
>
> In [33]: xm2 = xm.byteswap(True).newbyteorder()
>
> In [34]: (xm1 == xm2).all()
> True
>
> In [35]: xm1.dtype
> dtype('>f8')
>
> In [36]: xm2.dtype
> dtype('>f8')
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/6674/files/9d43cdbbba9de1991f935d60415ff4c586a7b8c6#r69373645>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AARy-B7EBkslE-LNO97zz923NnyiApzQks5qRgMWgaJpZM4JDmCM>
> .
>